### PR TITLE
feat(server/machine-learning): Configurable port

### PR DESCRIPTION
**Pull request by samip5**

### CODE REVIEW

**Comment by bo0tzz** [comment]
Since the applications are running in docker, why is this needed?

**Comment by samip5** [comment]
> Since the applications are running in docker, why is this needed?

Because not everyone is able to use Docker (nor the default ports), and I know it might not be officially recommended, but it would help those use cases.

**Comment by samip5** [comment]
The web container already is figuring out it's port from env, so I though it would make sense to do for others as well. Am I missing something here?

File: machine-learning/src/main.ts [hunk 1]
```diff
@@ -5,7 +5,11 @@ import { Logger } from '@nestjs/common';
 async function bootstrap() {
   const app = await NestFactory.create(AppModule);
 
-  await app.listen(3003, () => {
+  const listeningPort: number = isNaN(Number(process.env.MACHINE_LEARNING_PORT))
```
Comment (jrasm91) [inline machine-learning/src/main.ts:8]: `const port = Number(process.env.MACHINE_LEARNING) || 3003`;

If we do this change, we should update `app.config.ts` to include it as optional number type.

File: server/apps/immich/src/main.ts [hunk 2]
```diff
@@ -27,6 +27,8 @@ async function bootstrap() {
     app.enableCors();
   }
 
+  const serverPort: number = isNaN(Number(process.env.SERVER_PORT)) ? 3001 : Number(process.env.SERVER_PORT);
```
Comment (jrasm91) [inline server/apps/immich/src/main.ts:30]: Can you also change this to use the simplified syntax that only uses the env variable once?

Comment (samip5) [reply server/apps/immich/src/main.ts]: I have pushed changes for that, I also fixed it in microservices.

**Review by jrasm91 — APPROVED** [review]
Looks good to me. I don't think these will be necessary very often, but it doesn't hurt to make the port configurable via an env.

**Pull request merged**